pkg/geo/geomfn/binary_predicates.go

```diff
@@ -143,6 +146,13 @@ func Intersects(a geo.Geometry, b geo.Geometry) (bool, error) {
 	if !a.CartesianBoundingBox().Intersects(b.CartesianBoundingBox()) {
 		return false, nil
 	}
+	prep, other, err := preparedCache.Get(a.SpatialObject(), b.SpatialObject())
```

Comment:
as discussed earlier, we want this cache to be protected by a session variable / cluster setting. please follow something like https://github.com/cockroachdb/cockroach/pull/61723/-- this can be it's own commit / PR.

(ideally in the future, we will have fine tuning which is dictated by the session variable / cluster settings from the user)

----

i would probably make the cache an interface, in case we have different implementations of the cache:

```
type CacheResult struct {
   UseCache bool // there may be some cases where it may not be worth caching, e.g. if the geoms are small. this would've been part of experimentation but i'm tihnking we probably won't have time for that :)
   PreparedGeometry geos.PreparedGeometry
   OtherGeometry geo.Geometry
}
type Cache interface {
   Get(a geopb.SpatialObject, b geopb.SpatialObject) (CacheResult, error)
}
```
and have Intersects be something like
```
func Intersects(cache geosprepared.Cache, a geo.Geometry, b geo.Geometry) {
  // ... srid, bbox check ...
   if cache != nil {
      r, err := ....
      if r.UseCache { ... }
   }
   // 
}
```

then put the cache in builtins package, and call Intersects using the cache. we probably don't the session variable reading doesn't leak into this layer, so the builtins package would be responsible for passing the cache into here.